ndc-duckduckapi/src/handlers/query.ts

```diff
@@ -127,33 +168,57 @@ function build_where(
       }
       switch (expression.operator) {
         case "_eq":
-          sql = `${expression.column.name} = ?`;
-          break;
-        case "_like":
-          args[args.length - 1] = `%${args[args.length - 1]}%`;
-          sql = `${expression.column.name} LIKE ?`;
-          break;
-        case "_glob":
-          sql = `${expression.column.name} GLOB ?`;
+          sql = isTimestamp 
+            ? `${escape_double(expression.column.name)} = CAST(? AS TIMESTAMP)`
+            : integerType
+            ? `${escape_double(expression.column.name)} = CAST(? AS ${integerType})`
+            : `${escape_double(expression.column.name)} = ?`;
           break;
         case "_neq":
-          sql = `${expression.column.name} != ?`;
+          sql = isTimestamp 
+            ? `${escape_double(expression.column.name)} != CAST(? AS TIMESTAMP)`
```

Comment:
This could be refactored to pull the right hand side of the operator out into an expression which is computable only from the type, and reused. The left hand side seems to always  be the same.